token-manager/src/main.rs

```diff
@@ -0,0 +1,183 @@
+use std::{
+    env,
+    net::SocketAddr,
+};
+use axum::{
+    extract::{Query},
+    response::IntoResponse,
+    routing::post,
+    Router,
+};
+use log::{info, LevelFilter};
+use serde::{Deserialize, Serialize};
+use env_logger::Builder;
+use serde_json;
+use uuid::Uuid;
+use tokio::time::{sleep, Duration};
+use log::error;
+use lettre::message::header::ContentType;
+use lettre::transport::smtp::authentication::Credentials;
+use lettre::{Message, SmtpTransport, Transport};
+
+
+const MAX_RETRIES: u32 = 3;
+
+#[derive(Deserialize)]
+struct QueryParams {
+    email: String,
+    projects: String,       // Comma-separated values
+    bridgehead_ids: String, // Comma-separated values
```

Comment:
When you want a comment associated with a property or struct you can put a doc comment above it starting with 3 slashes like so
```suggestion
    /// Comma-separated values
    projects: String,
    /// Comma-separated values
    bridgehead_ids: String,
```
This also integrates with the LSP so it will show the comment when you hover over the item.
In the future you could also make a custom struct for comma separated which parses them directly in a custom deserialize implementation but its fine for now.